source/bson-corpus/binary-vector-subtype.md

```diff
@@ -0,0 +1,94 @@
+# BSON Binary Subtype 9 - Vector
+
+- Status: Pending
+- Minimum Server Version: N/A
+
+______________________________________________________________________
+
+## Abstract
+
+This document describes the addition of a new subtype to the Binary BSON type. This subtype is used for efficient
+storage and retrieval of vectors. Vectors here refer to densely packed arrays of numbers, all of the same type.
+
+## Motivation
+
+These representations correspond to the numeric types supported by popular numerical libraries for vector processing,
+such as NumPy, PyTorch, TensorFlow and Apache Arrow. Storing and retrieving vector data using the same densely packed
+format used by these libraries can result in up to 8x memory savings and orders of magnitude improvement in processing
+efficiency. Without this support, MongoDB will be at a competitive disadvantage to databases that do and our users will
+bear the additional cost of storing and processing vector data.
+
+`*` The early addition of the "Packed Bit" representation was to facilitate partnerships in the expanding market of
+Generative AI, specifically in Vector Quantization. Succinctly put, a Binary Quantized Vector is just a vector of 0s and
+1s (bits), but it is often represented as a list of uint8 (int in Python). So, for example, the vector `[255, 0]` would
+be shorthand for the 16 bit vector `[1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0]`.\
+The authors are well-aware of the inherent
+ambiguity here. This is a market-standard, unfortunately. Change is inevitable.
+
+## META
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Specification
```

Comment:
This spec would be benefited by including a diagram or model of the format itself. For example the decimal128 spec defines "structs" to represent the data format.

It would also be great to include code examples in Python that show the expected use cases (python list -> bson vector and vice versa).

Reply:
Happy to oblige, but I'm unsure what you have in mind for the model.

As to code examples, it was suggested to me not to include python code.  There's now a link in the reference implementation. Sufficient?

Reply:
We can either add an image, e.g. https://github.com/mongodb/specifications/blob/b5cbae0211d3d5049680cfc0803e4589474464bc/source/transactions/transactions.md?plain=1#L261, or ascii art (https://github.com/mongodb/specifications/blob/b5cbae0211d3d5049680cfc0803e4589474464bc/source/connection-string/connection-string-spec.md?plain=1#L32).  Here's an example of some implementation suggestion code: https://github.com/mongodb/specifications/blob/b5cbae0211d3d5049680cfc0803e4589474464bc/source/auth/auth.md#hostname-canonicalization

Reply:
This is over the top. How about an example:

For example, a vector [6, 7] of dtype PACKED_BIT (\x10) with a padding of 3 would like like this:
`b"\x10\x03\x06\x07'`: 1 byte for dtype, 1 for padding, and 1 for each uint8.

Reply:
@ShaneHarvey @blink1073 Please take a look now. I've added an html table.